Docs and yaml tests for viewing API key's limited-by (#89443)
elastic/elasticsearch: x-pack/docs/en/rest-api/security/get-api-keys.asciidoc

The 'realm_name' or 'username' parameters cannot be specified when this
parameter is set to 'true' as they are assumed to be the currently authenticated ones.

`with_limited_by`::
(Optional, Boolean) A boolean flag to return the snapshot of the owner user's role descriptors
associated with the API key. An API key's actual permission is the intersection of
its <<api-key-role-descriptors,assigned role descriptors>> and the owner user's role descriptors

> (effectively limited by it). An API key cannot retrieve any API key's limited-by role descriptors
> (including itself) unless it has `manage_api_key` or higher privileges.

Reviewer: Optional:

```suggestion
(effectively limited by it). An API key must have `manage_api_key` or higher privileges to retrieve the limited-by role descriptors of any API key, including itself.
```

A successful call returns a JSON structure that contains the information of the API key:

[source,js]
--------------------------------------------------
{
  "api_keys": [ <1>
    {
      "id": "VuaCfGcBCdbkQm-e5aOx", <2>
      "name": "my-api-key", <3>
      "creation": 1548550550158, <4>
      "expiration": 1548551550158, <5>
      "invalidated": false, <6>
      "username": "myuser", <7>
      "realm": "native1", <8>
      "metadata": { <9>
        "application": "myapp"
      },
      "role_descriptors": { }, <10>
      "limited_by": [  <11>
        {
          "role-power-user": {
            "cluster": [
              "monitor"
            ],
            "indices": [
              {
                "names": [
                  "*"
                ],
                "privileges": [
                  "read"
                ],
                "allow_restricted_indices": false
              }
            ],
            "applications": [ ],
            "run_as": [ ],
            "metadata": { },
            "transient_metadata": {
              "enabled": true
            }
          }
        }
      ]
    }
  ]
}
--------------------------------------------------
// NOTCONSOLE

> <1> The list of API keys that were retrieved for this request.

Reviewer: Nit preference but I would omit notes on the obvious stuff, like `id` and `name` -- it mostly just clutters the doc and it's harder to find the important info IMO.

Reviewer: Ah I see, it's a copy from below. Good with keeping as is, in that case.
